network/wsNetwork.go

```diff
@@ -1965,6 +1982,45 @@ func (wn *WebsocketNetwork) prioWeightRefresh() {
 	}
 }
 
+// This logic assumes that the relay address suffixes
+// correspond to the primary/backup network conventions. If this proves to be false, i.e. one network's
+// suffix is a substring of another network's suffix, then duplicates can end up in the merged slice.
+func (wn *WebsocketNetwork) mergePrimarySecondaryRelayAddressSlices(network protocol.NetworkID,
+	primaryRelayAddresses []string, secondaryRelayAddresses []string, dedupExp *regexp.Regexp) (dedupedRelayAddresses []string) {
+
+	if dedupExp == nil {
+		// No expression provided, so just append the slices without deduping
+		return append(primaryRelayAddresses, secondaryRelayAddresses...)
+	}
+
+	var relayAddressPrefixToValue = make(map[string]string, 2*len(primaryRelayAddresses))
+
+	for _, pra := range primaryRelayAddresses {
+		var normalizedPra = strings.ToLower(pra)
+
+		var pfxKey = dedupExp.ReplaceAllString(normalizedPra, "")
+		if _, exists := relayAddressPrefixToValue[pfxKey]; !exists {
+			relayAddressPrefixToValue[pfxKey] = normalizedPra
+		}
```

Comment:
I believe this logic is broken when the dedup pattern's <name> is not a prefix.

i.e. take `dedup=algorand.<name>.provider.net`
The expression would be `(algorand.provider.net)`
The `ReplaceAllString` would not match `algorand.relay1.provider.net`

Request:
assert that `<name>` is a prefix, or support infix names.

Reply:
Resolved in latest commit.